src/main/java/by/custom/utilcalculator/service/MessagesChecker.java

```diff
@@ -0,0 +1,50 @@
+package by.custom.utilcalculator.service;
+
+import by.custom.utilcalculator.controller.Bot;
+import by.custom.utilcalculator.repository.resources.Commands;
+import org.telegram.telegrambots.meta.api.methods.send.SendMessage;
+import org.telegram.telegrambots.meta.api.objects.Message;
+import org.telegram.telegrambots.meta.api.objects.Update;
+
+public class MessagesChecker {
+    private final MessagesCreator messagesCreator;
+    private MessagesChecker() {
+        messagesCreator = MessagesCreator.getInstance();
+    }
+
+    public static MessagesChecker getInstance() {
+        return MessagesCheckerHolder.MESSAGES_CHECKER;
+    }
+
+    public SendMessage checkMessageBeforeExecutionAndGetResult(Update update){
+        Message message = update.getMessage();
+        if (update.hasMessage() && message.hasText()) {
+            return getCheckInputMessageAndGetAnswer(message);
+        } else {
+            return messagesCreator.getSorryMessage(message);
+        }
+    }
+
+    public SendMessage getCheckInputMessageAndGetAnswer(Message message) {
+        String usersMessage = message.getText();
+        SendMessage sendMessage;
+        switch (usersMessage) {
+            case Commands.START -> sendMessage = messagesCreator.getGreetingMessage(message);
+            case Commands.EAES, Commands.OTHER_COUNTRIES -> sendMessage = Bot.getBot().getOriginCarMessage(message);
```

Comment:
It's not correct to call Bot methods from this class. Usually, lower-level classes don't know about higher. 
We will discuss it on call.

Reply:
I agree. Next pull request is about fixing this issue. I hope, I did everything OK